### ISSUE TITLE
Fiberize question

### ISSUE THREAD

**Issue by duclet** [xolvio/chimp#437]
I'm trying to get Mocha's "retries" functionality to work. However, it is not working due to the fact that the exception is caught and then thrown in the nextTick function in the fiberize util. If I just get rid of the try-catch block entirely, it seems to work. I've all of my tests and they seem to work so it looks like the try-catch block isn't really necessary so I'll probably do that for my forked version of this. However, I do have a question on why it was setup as such. Was there any particular reason why? For reference, here is the file in question:
https://github.com/xolvio/chimp/blob/master/src/lib/utils/fiberize.js#L17

**Comment by samhatoum** [comment]
Great question. @Sanjo was the original author, so maybe he can chime in?

**Comment by Sanjo** [comment]
I don't know the reason. Try this version:

```js
export function fiberize(fn) {
  return function (done) {
    const self = this;
    Fiber(function () {
      try {
        if (fn.length === 1) {
          fn.call(self, done);
        } else {
          fn.call(self);
          done();
        }
      } catch (error) {
        done(error);
      }
    }).run();
  };
}
```

**Comment by samhatoum** [comment]
@duclet let us know and we can merge it if it fixes your issue

**Comment by duclet** [comment]
Yeah, that was what I ended up changing my fork to so it does work.

**Comment by samhatoum** [comment]
Let us know how you get on @duclet, I'd like to merge this in

**Comment by duclet** [comment]
Feel free to merge it in, it works.

**Comment by duclet** [comment]
Or are you wanting me to create a pull request?

**Comment by samhatoum** [comment]
No it's all good, we'll do it next. Thanks

**Comment by lucetius** [comment]
Merged with #463
